src/czbenchmarks/datasets/dataset.py

```diff
@@ -0,0 +1,90 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import IO, Any
+import os
+
+from .types import Organism
+
+
+class Dataset(ABC):
+    """
+    A base class for task-specific datasets. Each concrete dataset class will extract the data required for a specific type of task from the provided input file. These task specific data items can then be retrieved as object instance variables or written to files for later use.
+    
+    path: Path
+
+    Args:
+        ABC (_type_): _description_
+    """
+    
+    name: str # TODO: rename to dataset_type?
+    path: Path
+    dir: Path
+    organism: Organism
+    
+    
+    def __init__(self, name: str, path: Path, organism: Organism, **kwargs: Any):
+        self.name = name
+        
+        self.path = path
+        if not os.path.exists(self.path):
```

Comment:
since `path` is a `Path`, you can do `if not self.path.exists()`. Similarly for `is_dir` below

https://docs.python.org/3/library/pathlib.html#corresponding-tools